dev/breeze/src/airflow_breeze/utils/python_versions.py

```diff
@@ -45,14 +45,12 @@ def get_python_version_list(python_versions: str) -> list[str]:
     return python_version_list
 
 
-def check_python_version():
-    error = False
-    if not sys.version_info >= (3, 9):
-        get_console().print("[error]At least Python 3.9 is required to prepare reproducible archives.\n")
-        error = True
-    if error:
+def check_python_version(release_provider_packages: bool = False):
+    if not sys.version_info < (3, 12) and release_provider_packages:
```

Comment:
Can you explain more why it's neeeded? ? I don't understand why we can't run `breeze` commands from 3.12.

Cos `dist/apache_airflow_providers_apache_beam-5.8.1-py3-none-any.whl` got built for me without any problems on 3.12

Reply:
I would assume the answer is in https://github.com/apache/airflow/pull/42944/files#diff-53e7c91dbd78398c4d785e8b277e326c04cc5695b0eb01297d9f79b37e53e2f3R75

Reply:
Whl yes. But part of release managament also (as explained in detail in linked PRs and commits - part of relase management provider preparation is to take .sdist packages and install them. See https://github.com/apache/airflow/pull/42936#issuecomment-2407318919 where I explained it in detail and linke dto the PR that introduced the change.

I think also it saves a lot of time of people for additional explaination when people who review and comment PRs actually look at the mesages linked PRs and explanation especialy when the links are explicitly mentioned as the source of information and are explicitly linked in the commit message. I think it avoids a lot of duplication of the same information and requires very little effort from person who is reviewing PRs, especailly if those comments are made on PRs of those pepple. This allows to save time on multiple explanation at the expense of extra clicking a link that contains detailed explanation.

It's just pretty tiring to have to explain the same things multiple times to the same person.

Reply:
For example @ashb. If you would do once verification of the providers as a PMC member, you would immediately see that Python 3.12 will fail in this step:

<img width="1110" alt="Screenshot 2024-10-11 at 21 29 02" src="https://github.com/user-attachments/assets/144b3ccb-32a1-4901-8d0c-dfca14ffe9cc">

This is necesary step to check reproducibility of building the packages.

But as far as I see you've not done it recently so you might have not realized that this is happening.

Reply:
BTW this will only fal when verifying "apache-beam" provider - so following the common-sense approach "fail early rather than at the end of process, It makes absolutely sense to fail it during the check.

Since you rarely (or never) run provider's verification, you should not be impacted.

Reply:
It's not verification of the releases that this caused me issues on -- it's working on the deps of Airflow that this caused me a lot of hassle, and in order to run the CI locally I needed to build wheel packages of all the providers.